packages/cloud-cognitive/src/components/InlineEditV2/InlineEditV2.js

```diff
@@ -0,0 +1,214 @@
+import React, { useState, useEffect, forwardRef, useRef } from 'react';
+import { Button } from 'carbon-components-react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import {
+  Edit24,
+  Checkmark24,
+  Close24,
+  EditOff24,
+  WarningFilled24,
+} from '@carbon/icons-react';
+import { pkg, carbon } from '../../settings';
+
+const componentName = 'InlineEditV2';
+const blockClass = `${pkg.prefix}--inline-edit-v2`;
+
+export let InlineEditV2 = forwardRef(
+  (
+    {
+      cancelLabel,
+      editLabel,
+      invalid,
+      invalidLabel,
+      onCancel,
+      onChange,
+      onSave,
+      readOnly,
+      saveLabel,
+      value,
+      ...rest
+    },
+    ref
+  ) => {
+    const [focused, setFocused] = useState(false);
+    const [initialValue, setInitialValue] = useState('');
+    const [dirtyInput, setDirtyInput] = useState(false);
+    const inputRef = useRef(null);
+    const canSave = value !== initialValue && !invalid;
+
+    useEffect(() => {
+      if (!initialValue && !dirtyInput) {
+        setInitialValue(value);
+      }
+    }, [initialValue, dirtyInput, value]);
+
+    const isTargetingChild = ({ currentTarget, relatedTarget }) =>
+      currentTarget.contains(relatedTarget);
+
+    const onChangeHandler = ({ target }) => {
+      if (!dirtyInput) {
+        setDirtyInput(true);
+      }
+
+      onChange(target.value);
+    };
+
+    const onBlurHandler = (e) => {
+      if (readOnly) {
+        return;
+      }
+
+      if (!isTargetingChild(e)) {
+        if (canSave) {
+          onSaveHandler();
+        } else {
+          onCancelHandler();
+        }
+      }
+    };
+
+    const onFocusHandler = (e) => {
+      if (readOnly) {
+        return;
+      }
+
+      if (!isTargetingChild(e)) {
+        inputRef.current.focus();
+        setFocused(true);
+      }
+    };
+
+    const onSaveHandler = (resetFocus = true) => {
+      // when return or esc keys are used blur isn't being called, so don't reset the focus
+      if (resetFocus) {
+        setFocused(false);
+      }
+
+      setInitialValue(value);
+      setDirtyInput(false);
+      onSave();
+    };
+
+    const onCancelHandler = (resetFocus = true) => {
+      if (resetFocus) {
+        setFocused(false);
+      }
+
+      setDirtyInput(false);
+      onCancel(initialValue);
+    };
+
+    const returnHandler = () => {
+      if (canSave) {
+        onSaveHandler(false);
+      }
+    };
+
+    const escapeHandler = () => {
+      onCancelHandler(false);
+    };
+
+    const onKeyHandler = (e) => {
+      switch (e.keyCode) {
```

Comment:
We should probably use `event.key` since `keyCode` is deprecated. See example [here](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/main/packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/InlineEdit/handleGridKeyPress.js#L125)

Reply:
was unaware. thanks for the head's up!